### PR TITLE
Fix some golint issues and add its dependencies

### LINKED ISSUE #250: Fix go lint issues highlighted in Go report Card
Go Report Card [Results](https://goreportcard.com/report/github.com/vmware-tanzu/cluster-api-provider-bringyourownhost) for go lint is only 56%.  Most of these are for exported fields/methods not having comments. We need to fix these lint issues.

### PR DESCRIPTION
**Pull request by jayesh-srivastava**
**What this PR does / why we need it**:
This PR fixes some non-comment related golint issues and also adds its dependencies.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
This PR partially fixes #250 

**Special notes for your reviewer**
This PR only fixes some of the golint issues. A majority of the issues were related to structs and functions not being commented which is why I'm unable to work on those as I don't have a clear understanding of the whole codebase atm.
P.S - [cluster-api-provider-bringyourownhost/issues/250#issuecomment-1037227086](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/250#issuecomment-1037227086) and following comments.

### CODE REVIEW

**Comment by jayesh-srivastava** [comment]
cc @anusha94

File: test/e2e/e2e_docker_helper.go [hunk 1]
```diff
@@ -17,8 +17,7 @@ import (
 	"github.com/docker/docker/api/types/network"
 	"github.com/docker/docker/client"
 	"github.com/docker/docker/pkg/archive"
-	"github.com/docker/docker/pkg/system"
-	. "github.com/onsi/gomega" // nolint: stylecheck
+	"github.com/docker/docker/pkg/system" // nolint: stylecheck
```
Comment (anusha94) [inline test/e2e/e2e_docker_helper.go:20]: why the `nolint` here?

**Review by anusha94 — COMMENTED** [review]
minor nit.

File: test/e2e/e2e_docker_helper.go [hunk 2]
```diff
@@ -17,8 +17,7 @@ import (
 	"github.com/docker/docker/api/types/network"
 	"github.com/docker/docker/client"
 	"github.com/docker/docker/pkg/archive"
-	"github.com/docker/docker/pkg/system"
-	. "github.com/onsi/gomega" // nolint: stylecheck
```
Comment (anusha94) [inline test/e2e/e2e_docker_helper.go:21]: Removing `gomega` is causing all sorts of lint and test errors I believe.

File: test/e2e/e2e_docker_helper.go (hunk 1 above)
Comment (jayesh-srivastava) [reply test/e2e/e2e_docker_helper.go:20]: @anusha94 This comment line was already there in the codebase.

File: test/e2e/e2e_docker_helper.go (hunk 2 above)
Comment (jayesh-srivastava) [reply test/e2e/e2e_docker_helper.go:21]: The no lint comment was used for gomega but that's an unused import and vscode itself removes it while saving the file.

**Comment by jayesh-srivastava** [comment]
@anusha94 I guess the issues are related to linters. Also the golangci-lint linters aren't in parity with CAPI which should be the case because it helps in making the developer experience same. I've seen these issues for other CAP* projects like CAPZ and CAPG. I'll open the same for CAP-BYOH as well. After closing that issue, I guess this should work.

**Pull request closed without merging**